### PR TITLE
Fixed routing issue caused by JS state management

### PR DESCRIPTION
**Pull request by tehuanmelo**

### CODE REVIEW

**Review by Taanviir — CHANGES_REQUESTED** [review]
I have looked at the code and tested the changes however there was an issue where any link other than `localhost` threw a 404 error if entered in the URL search bar.

File: backend/static/scripts/main.js [hunk 1]
```diff
@@ -50,36 +51,39 @@ function handleSpaLinkEvent(target) {
...
+        if (pageTitle === 'Login') history.replaceState(null, '', '/users/login/');
+        else if (pageTitle === 'Home') history.replaceState(null, '', '/');
+    }
+    else {
+        const oldUrl = document.location.pathname;
+        if (oldUrl !== url) history.pushState(null, '', url);
+    }
+  
     const newTitle = page.querySelector('title').innerHTML;
+    document.title = '';
```
Comment (Taanviir) [inline backend/static/scripts/main.js:72]: also why is this being done

Comment (tehuanmelo) [reply backend/static/scripts/main.js:72]: it must be there, to prevent caching

**Comment by tehuanmelo** [comment]
> I have looked at the code and tested the changes however there was an issue where any link other than `localhost` threw a 404 error if entered in the URL search bar.

Issue fixed.
Please review it and let me know if the error persists.

**Review by Taanviir — APPROVED** [review]
Everything looks good, you can go ahead and merge

**Pull request merged**